system/lib/webgpu/webgpu.cpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2023 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+/*
+ * This file and library_webgpu.js together implement <webgpu/webgpu.h>.
+ */
+
+#include <webgpu/webgpu.h>
+
+extern "C" {
```

Comment:
This file looks like its all C code so maybe skip the extern "C" stuff here and just make this into a `.c` file?